frontend/catalyst/pipelines.py

```diff
@@ -37,6 +38,42 @@
 from catalyst.utils.exceptions import CompileError
 
 
+class KeepIntermediateLevel(enum.IntEnum):
+    """Enum to control the level of intermediate file keeping."""
+
+    NONE = 0  # No intermediate files are kept.
+    PIPELINE = 1  # Intermediate files are saved after each pipeline.
+    PASS = 2  # Intermediate files are saved after each pass.
+
+
+def _parse_keep_intermediate(
+    level: Union[str, int, bool, KeepIntermediateLevel, None],
+) -> KeepIntermediateLevel:
+    """Parse the keep_intermediate value into a KeepIntermediateLevel enum."""
+    if isinstance(level, KeepIntermediateLevel):
+        return level
+    elif level is None:
+        return KeepIntermediateLevel.NONE
+    elif isinstance(level, bool):
+        return KeepIntermediateLevel.PIPELINE if level else KeepIntermediateLevel.NONE
+    elif isinstance(level, int):
+        try:
+            return KeepIntermediateLevel(level)
+        except ValueError as e:
+            raise ValueError(
+                f"Invalid int for keep_intermediate: {level}. " "Valid integers are 0, 1, 2."
+            ) from e
+    elif isinstance(level, str):
+        try:
+            return KeepIntermediateLevel[level.upper()]
+        except KeyError as e:
+            raise ValueError(
+                f"Invalid string for keep_intermediate: {level}. "
+                "Valid strings are 'none', 'pipeline', 'pass'."
+            ) from e
+    raise TypeError(f"Invalid type for keep_intermediate: {type(level)}.")
```

Comment:
I think this would be simpler if you just did a `match` and `case` for 0, 1, 2. Everything else just raises an error.

```
Python 3.12.3 (main, Feb  4 2025, 14:48:35) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> True == 1
True
>>> False == 0
True
>>> 
```

E.g.,

```suggestion
    """Parse the keep_intermediate value into a KeepIntermediateLevel enum."""
    match level:
      case 0 | 1 | 2:
        # do your success logic
      case _:
        # do your fail logic
```